frontend/src/metabase/query_builder/actions/navigation.ts

```diff
@@ -197,15 +205,17 @@ export const updateUrl = createThunkAction(
         (locationDescriptor.hash || "") === (window.location.hash || "");
       const isSameCard =
         currentState && isEqualCard(currentState.card, newState.card);
-      const isSameMode =
-        getQueryBuilderModeFromLocation(locationDescriptor).mode ===
-        getQueryBuilderModeFromLocation(window.location).mode;
 
       if (isSameCard && isSameURL) {
         return;
       }
 
       if (replaceState == null) {
+        const isSameMode =
```

Comment:
this change caused a broken test in `navigation.ts`, so we needed to go back one more time in the test